### PR TITLE
Split windows pyside/pyqt into own GitHub action check

### PR DESCRIPTION
**Pull request by Carreau**
I find it relatively confusing in the github action UI to find issues on
windows that happen only on Pyqt/pyside, I regularly forget that the
windows action run both, and I scroll to find the error for a minute or
so before figuring out that the I need to expand the other section.

I also find that having from the GitHub UI indication of which
variant(s) failed is a good way to guess wether we have a random failure
or an actual issue.

I'm hoping this may also reduce the "time to CI results" as we can now
run pytest/pyqt in parallel.

-- 

At least I hope that how we split the tox envs, I'm not too sure. It also look like there is no "headless" testing on windows, so I did not include it.

### CODE REVIEW

**Comment by tlambert03** [comment]
fwiw, the ubuntu actions also run both (basically, when you run `tox` it will run all matching [factors](https://tox.readthedocs.io/en/latest/config.html#tox-environments) in the [envlist](https://tox.readthedocs.io/en/latest/config.html#generating-environments-conditional-settings)... 
I get that windows is the one that's failing the most, so it's more obvious there, but just wanted to point that out.

I think the main reason we were doubling up is to reduce the total number of github actions jobs we're running.  We _do_ sometimes see queuing already happening, but I don't totally understand how github decides that for a given org... and it's been mentioned that we might purchase additional CI time.

**Comment by Carreau** [comment]
> I think the main reason we were doubling up is to reduce the total number of GitHub actions jobs we're running.

Yes, I understand that, in particular there seem to be a large setup overhead (running just pyside or pyqt take 11-12min, and running both, 15-16).

I would be ok, if we decide to not go with this approach. We can also do some work of speeding up test-suite if CI is problematic. I'm sure we can likely mark some of the slowest tests with `@slow` and skip them on some platform.

Maybe the install step can be faster as well, don't know if we can cache tox envs, but it's ~4min/action of installing.

**Comment by sofroniewn** [comment]
I don't have strong opinions here so am happy to defer to whatever group thinks is best. I can also prioritize getting some CI credits for the project. Haven't done that yet, but should!!

**Comment by Carreau** [comment]
I'm not convinced we need CI hours, it's easy to throw more time at the problem, but I believe that faster tests is always good for contributors even when testing on local machines.

**Comment by tlambert03** [comment]
definitely agree there.  I guess I rarely care about this though precisely _because_ I test locally most of the time.  Generally running a subset of the tests

**Comment by tlambert03** [comment]
still want to try this? i'm ok with it.  shall we merge?

**Review by sofroniewn — APPROVED** [review]
> still want to try this? i'm ok with it. shall we merge?

I like it. Looks like windows 3.8 is failing right now on main, so this might help debug. I'll merge now, if unhelpful we can always revert

**Comment by Carreau** [comment]
Thanks !

**Pull request merged**
